integration-tests/test_registration.py

```diff
@@ -0,0 +1,70 @@
+import os
+import subprocess
+
+import pytest
+
+MACHINE_ID_FILE: str = "/etc/insights-client/machine-id"
+
+
+def test_machineid_only_when_registered(external_candlepin, subman, insights_client, test_config):
+    """`machine-id` is only present when insights-client is registered."""
+    assert not insights_client.is_registered
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+    res = insights_client.run(check=False)
+    assert "This host has not been registered. Use --register to register this host." in res.stdout
+    assert not insights_client.is_registered
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+    subman.register(
```

Comment:
there is a session scoped fixture in conftest that registers subman at the beginning of test, you dont need to reregister. is its needed for some specific reason?

Reply:
The test plan for CCT-161 included sub-man registrations, that's why I copied them over. Now that I'm reading it again, the RHSM registration doesn't change anything, we can start with it already being registered.